Bottom: f49b46ce265a0adae2faf1f0f1b90c51f752861e
Top:    bdd79cf0331634ab1059166f6ab1dacf10e6f822
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-14 15:52:13 +0800

Refresh of autofs-5.0.7-check-for-protocol-option.patch

---

diff --git a/modules/mount_nfs.c b/modules/mount_nfs.c
index 9de8a73..3d2ccea 100644
--- a/modules/mount_nfs.c
+++ b/modules/mount_nfs.c
@@ -156,6 +156,12 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 					if (port < 0)
 						port = 0;
 					port_opt = cp;
+				} else if (strncmp("proto=udp", cp, o_len) == 0 ||
+					   strncmp("udp", cp, o_len) == 0) {
+					vers &= ~TCP_SUPPORTED;
+				} else if (strncmp("proto=tcp", cp, o_len) == 0 ||
+					   strncmp("tcp", cp, o_len) == 0) {
+					vers &= ~UDP_SUPPORTED;
 				}
 				/* Check for options that also make sense
 				   with bind mounts */
@@ -167,6 +173,10 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 			}
 		}
 
+		/* In case both tcp and udp options were given */
+		if ((vers & NFS_PROTO_MASK) == 0)
+			vers |= NFS_PROTO_MASK;
+
 		debug(ap->logopt, MODPREFIX
 		      "nfs options=\"%s\", nobind=%d, nosymlink=%d, ro=%d",
 		      nfsoptions, nobind, nosymlink, ro);
